iothub/service/src/AmqpServiceClient.cs

```diff
@@ -235,6 +235,16 @@ public override Task<CloudToDeviceMethodResult> InvokeDeviceMethodAsync(string d
 
         public override Task<CloudToDeviceMethodResult> InvokeDeviceMethodAsync(string deviceId, string moduleId, CloudToDeviceMethod cloudToDeviceMethod, CancellationToken cancellationToken)
         {
+            if (string.IsNullOrWhiteSpace(deviceId))
+            {
+                throw new ArgumentException("Value should be non null and non empty", nameof(deviceId));
```

Comment:
Please switch to `ArgumentNullException` and omit the text message keeping just the argument. That allows the dotnet stack to choose the proper text string in any language and keeps the resulting binary smaller.

Reply:
I thought, I'll do it consistent to the other methods within the same class, which also throw `ArgumentException`. But I agree it is better to throw `ArgumentNullExcption`, so I changed it also in the other methods in the same class.
While scanning through the class I've found two places which do not specify ConfigureAwait, which I also fixed now.